internal/core/application/feeder/service.go

```diff
@@ -0,0 +1,175 @@
+package feeder
+
+import (
+	"context"
+	"fmt"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/tdex-network/tdex-daemon/internal/core/domain"
+
+	"github.com/tdex-network/tdex-daemon/internal/core/ports"
+)
+
+type Service struct {
+	repoManager ports.RepoManager
+	feederSvc   ports.PriceFeeder
+}
+
+func NewService(
+	feederSvc ports.PriceFeeder,
+	repoManager ports.RepoManager,
+) (Service, error) {
+	svc := Service{
+		repoManager: repoManager,
+		feederSvc:   feederSvc,
+	}
+
+	if err := svc.startPriceFeed(feederSvc); err != nil {
+		return Service{}, err
+	}
+
+	return svc, nil
+}
+
+func (s *Service) Stop(ctx context.Context) {
+	s.feederSvc.Stop(ctx)
+}
+
+func (s *Service) StartFeed(ctx context.Context, feedID string) error {
+	priceFeed, err := s.feederSvc.GetPriceFeed(ctx, feedID)
+	if err != nil {
+		return err
+	}
+
+	market, err := s.repoManager.MarketRepository().GetMarketByAssets(
+		ctx,
+		priceFeed.GetMarket().GetBaseAsset(),
+		priceFeed.GetMarket().GetQuoteAsset(),
+	)
+	if err != nil {
+		return err
+	}
+
+	if !market.IsStrategyPluggable() {
+		return fmt.Errorf("market is not pluggable")
+	}
+
+	return s.feederSvc.StartFeed(ctx, feedID)
+}
+
+func (s *Service) StopFeed(ctx context.Context, feedID string) error {
+	return s.feederSvc.StopFeed(ctx, feedID)
+}
+
+func (s *Service) AddPriceFeed(
+	ctx context.Context, market ports.Market, source, ticker string,
+) (string, error) {
+	return s.feederSvc.AddPriceFeed(ctx, market, source, ticker)
+}
+
+func (s *Service) UpdatePriceFeed(
+	ctx context.Context, id, source, ticker string,
+) error {
+	priceFeed, err := s.feederSvc.GetPriceFeed(ctx, id)
+	if err != nil {
+		return err
+	}
+
+	if priceFeed.IsStarted() {
+		return fmt.Errorf("feed needs to be stopped before updating it")
+	}
+
+	return s.feederSvc.UpdatePriceFeed(ctx, id, source, ticker)
+}
+
+func (s *Service) RemovePriceFeed(ctx context.Context, id string) error {
+	priceFeed, err := s.feederSvc.GetPriceFeed(ctx, id)
+	if err != nil {
+		return err
+	}
+
+	if priceFeed.IsStarted() {
+		return fmt.Errorf("feed needs to be stopped before updating it")
+	}
+
+	market, err := s.repoManager.MarketRepository().GetMarketByAssets(
+		ctx,
+		priceFeed.GetMarket().GetBaseAsset(),
+		priceFeed.GetMarket().GetQuoteAsset(),
+	)
+	if err != nil {
+		return err
+	}
+
+	if market.IsStrategyPluggable() {
+		return fmt.Errorf("cant remove pluggable market")
+	}
+
+	return s.feederSvc.RemovePriceFeed(ctx, id)
+}
+
+func (s *Service) GetPriceFeed(
+	ctx context.Context, baseAsset, quoteAsset string,
+) (ports.PriceFeed, error) {
+	return s.feederSvc.GetPriceFeedForMarket(ctx, baseAsset, quoteAsset)
+}
+
+func (s *Service) ListSources(ctx context.Context) []string {
+	return s.feederSvc.ListSources(ctx)
+}
+
+func (s *Service) ListPriceFeeds(ctx context.Context) ([]ports.PriceFeed, error) {
+	return s.feederSvc.ListPriceFeeds(ctx)
+}
+
+func (s *Service) listenPriceFeedChan(priceFeedChan chan ports.PriceFeedChan) {
```

Comment:
Not sure having a single channel for all price feeds is the best design for this feature mostly because the daemon this way starts a go routine to listen to a channel that potentially can never be used by the operator, if, for example, he never sets up a price feed for any market. Sounds a bit like wasting resources somehow.

I do think it would be better having a dedicated channel for every price feed, this way we could:
- start a go routine to listen to a channel only once the price feed is started
- stop a go routine that listens to the channel right after the price feed is stopped

With this design, at app level we end up having a go routine for each active price feed. If there are no price feeds, the daemon doesn't perform any background job at all.

Reply:
If concern are resources in case of un-used chan i dont see this as issue, cause this is insignificant impact, initial go routine alloc is ~2KB.
If the concern is processing speed(and potential wait on chan "unlock") of price feed when received on chan we can put db update in go routine(talking about code in [here](https://github.com/sekulicd/tdex-daemon/blob/cfa1ec5dbaf46930f2f6df7a4f39c99bd55ca795/internal/core/application/feeder/service.go#L130) and maybe update to buffered chan.
Adding multiple chans would add more complexity to already bit complex design and i think redesign of both pkg's and infra feeder would be needed.
My suggestion, is to continue with current approach, i tested it pretty solid and seems like working fine, if we want to optimise for performance or reason about new design maybe its better to move to another issue as this is already big PR.